Automattic/jetpack
# AI Assistant: Add Jetpack AI module and disable all features when the module is disabled

Fixes #38461
Fixes #37059

## Proposed changes:

* Removes the "New" chip on the AI Assistant entry on the settings page
* Adds a Jetpack AI module
* Reacts to changes on the Jetpack AI module enabled value and activates or deactivates the AI features
* Do not enqueue assets if the module was disabled at page load

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Check that the AI extensions are working as expected
* Open another tab and go to `wp-admin/admin.php?page=jetpack_modules`
* Find Jetpack AI and disable it
* Refresh the editor (the change from another tab is not reflected on this tab's state)
* Check that all Jetpack AI features are disabled. That includes the AI Assistant block, paragraph, heading, list, form, logo and image extensions, featured image and the sidebar features, as the sidebar itself is removed
* Go to `wp-admin/admin.php?page=jetpack#/writing`
* Enable Jetpack AI by clicking on `Add Jetpack AI to your site and post editors` 
* Refresh the editor again
* Check that Jetpack AI features are enabled
* On the browser's console, use: `wp.data.dispatch('jetpack-modules').updateJetpackModuleStatus( { name: 'ai', active: false } )`
* Check that the AI features are disabled without a need for a page refresh, except for the AI Assistant block
* Use `wp.data.dispatch('jetpack-modules').updateJetpackModuleStatus( { name: 'ai', active: true } )`
* Check that the AI features are enabled without a need for a page refresh
* Disable the module again and reload the page
* Use `wp.data.dispatch('jetpack-modules').updateJetpackModuleStatus( { name: 'ai', active: true } )`
* Check that the features are *not* enabled, as the assets were not loaded
* Reload the page
* Check that the AI features are enabled

![Screenshot from 2024-12-12 16-09-12](https://github.com/user-attachments/assets/9e4cfc5e-ce6e-4760-8905-76a3a7c10550)
![Screenshot from 2024-12-12 16-08-57](https://github.com/user-attachments/assets/a45ff8ba-ed43-41b1-a653-95cf8b8c5822)
